net/initiate.go

```diff
@@ -59,7 +60,7 @@ func (h *host) Initiate(who peer.AddrInfo, msg *SendKeysMessage, s common.SwapSt
 		stream:    stream,
 	}
 
-	go h.handleProtocolStreamInner(stream, s, make([]byte, messageBufferSize))
```

Comment:
the reason 1 buffer was allocated per peer is because allocations are expensive, on prev projects allocating buffers for reading network messages used a LOT of CPU. should update this in a follow up imo so the network has a buffer pool

Reply:
I was dealing with bugs and I was just shooting for simple.  I'm not opposed to moving back to a fixed size buffer.  For the non DDOS case, we are only allocating the needed message size (verses the max message size above). The go runtime (like libc) pre-allocates pools of smaller sized buffers. The CPU used for the volume of messages we exchange should be so small that you would have difficulty measuring it (verses noise from everything else we do).